readme: update readme (#303)
decred/dcrdex: README.md

## Using Decred DEX

**The Decred DEX is in early stages of development, and should not be used to
conduct trades on mainnet.** For those who would like to contribute, or just
poke around and offer feedback, there are a number of ways to do that.

- [Run **dcrdex** and **dexc** on simnet](../../wiki/Simnet-Testing). Recommended for development.
- [Run **dexc** on testnet](../../wiki/Testnet-Testing). Recommended for poking around.
- [Run the test app server](../../wiki/Test-App-Server). Useful for GUI development, or just to try everything out without needing to create wallets or connect to a **dcrdex** server.

## Server Installation

#### Dependencies

1. Linux
2. [Go >= 1.13](https://golang.org/doc/install)
3. [PostgreSQL 11+](https://www.postgresql.org/download/), [tuned](https://pgtune.leopard.in.ua/) and running.

#### Set up the database

In a PostgreSQL `psql` terminal, run

```
CREATE USER dcrdex WITH PASSWORD 'dexpass';
CREATE DATABASE dcrdex_testnet OWNER dcrdex;
```

#### Generate a master public key

The master public key is used for collecting registration fees. This can be
done with [dcrwallet](https://github.com/decred/dcrwallet) using the

> `getmasterpubkey` RPC command. Place the pubkey string into a new DEX

Reviewer: Make it obvious that it mustn't be the default account's master pub?
```suggestion
`getmasterpubkey ("account")` RPC command. Place the pubkey string into a new DEX
```

Reviewer: Also, L105 below reads `# Mainnet extended pubkey`, should probably be `testnet extended pubkey` or just `extended pubkey` as we've earlier discouraged running on mainnet.

limit orders and an order book. Settlement occurs on-chain. DEX's epoch-based
matching algorithm and rules of community conduct ensure that the order book
action you see is real and not an army of bots.

Trades are performed directly between clients through on-chain contracts with no
actual reliance on DEX, though clients do report their swap details both as a
courtesy and to prove compliance with trading rules. Trades are settled with
pure 4-transaction atomic swaps and nothing else. Because DEX collects no
trading fees, there's no intermediary token and no fee transactions.

Although trading fees are not collected, DEX does require a one-time
registration fee to be paid on-chain. Atomic swap technology secures all trades,
but client software must still adhere to a set of policies to ensure orderly
settlement of matches. Loss of registration fee is the heaviest consequence for
clients that break the rules.

## Contents

1. [Market API](#market-api)
1. [Client Applications and the Core Package](#client-applications-and-the-core-package)
1. [Using Decred DEX](#using-decred-dex)
1. [Server Installation](#server-installation)
1. [Client Installation](#client-installation)

> 1. [Contribute](#contribute)

Author: It doesn't look that great with bullets, imo, but it's probably better that way.

> 1. [Contribute](#contribute)

Reviewer 2: I think it looks pretty.  TBH looking at the numbered version at first had me looking for the number in the heading after click.  Maybe that's just me.

pgpass=dexpass
```

> The *~/.dcrdex/* is the default **app data directory** location used by the

Reviewer: ```suggestion
*~/.dcrdex* is the default **app data directory** location used by the
```

DEX server, but can be customized with the `--appdata` command-line argument.

Master public keys are network-specific, so **dcrd** and **dcrwallet** will need

> to be run on the same network on which the server will be run.

Reviewer: For dcrdex server setup, technically don't need **dcrwallet** to be running, at least not on the same machine as the server. I'm thinking a cold wallet is a better idea for collecting fees.

#### Run your asset daemons.

As of writing, only `dcrd`, `bitcoind`, and `litecoind` are supported. The
`txindex` configuration option must be set.

> #### Create the asset and market configuration file

Reviewer: Forgive the nitpicking
```suggestion
#### Create the assets and markets configuration file
```

Author: I'm not sure about this. I don't think I would call it an "assets configuration file", I think I would call it an "asset configuration file", even if it configures more than one asset. Same with markets.

Author: > Forgive the nitpicking

Forgive? I encourage it!

> A sample is given at *dcrdex/cmd/dcrdex/sample-markets.json*. See the

Reviewer: Could this be linked?

[**Per-asset Variables**](spec/admin.mediawiki) section of the specification for
more information on individual options.

#### Build and run dcrdex

From a command prompt, navigate to **server/cmd/dcrdex**. Build the executable
by running `go build`. The generated executable will be named **dcrdex**. Run

> `./dcrdex --help` to see configuration options that can be set either here or

Reviewer: ```suggestion
`./dcrdex --help` to see configuration options that can be set either in command line or
```

./dexc --testnet
```

Connect to the client from your browser at `localhost:5758`.

## Contribute

**Looking to contribute? We need your help** to make DEX &#35;1.

> Nearly all development is done in Golang and JavaScript. Work is coordinated

Reviewer 2: Would prefer to go with the proper name "Go" https://tip.golang.org/doc/faq#go_or_golang

> through [the repo issues](../../issues), so that's the best place to start.

Reviewer 2: Why not put the full URL here so that it opens to the right place no matter where the md is being viewed.
